### DISCUSSION TITLE
How to detect that SafariView has opened?

### DISCUSSION THREAD

**Discussion by lukebrandonfarrell** [wix/Detox#3019] [Q&A] [answered]
I'm looking to see what would be the process to assert that SafariView has opened i.e. clicking the link worked and it didn't just do nothing. I have tried `expect().toNotBeVisible()` on the view behind the SafariView, but it doesn't work. Any ideas?

**Comment by d4vidi** [comment]
@asafkorem any ideas of the top of your head, maybe?

**Comment by asafkorem** [comment, marked as answer]
@lukebrandonfarrell, can you help me by providing a code sample for the scenario that you tried to test?

The `SafariView` is a `SFSafariViewController` presented over the visible view controller. By "the view behind the SafariView" do you mean this VC's view? 
Have you tried to set and access it using the `testID` (`accessibilityidentifier`) of the view (`element(by.id('...'))`), or by its type (`element(by.type('SFSafariView'))`)?

**Reply by lukebrandonfarrell** [reply]
@asafkorem Will give this a try and get back to you, problem is that I just didn't know where to start for accessing `SFSafariView`

**Reply by lukebrandonfarrell** [reply]
@asafkorem It worked! You are the best! And thanks also @d4vidi

**Reply by asafkorem** [reply]
@lukebrandonfarrell hooray, glad to be of help! 🎉
